apps/sap-commerce-cloud/frontend/src/components/Dialog/Dialog.tsx

```diff
@@ -42,14 +44,18 @@ const Dialog: React.FC<DialogProps> = ({ sdk }) => {
   const load = useCallback(async () => {
     const { products, errors } = await fetchProductList(
       baseSite,
-      query,
+      debouncedQuery,
       page,
       sdk.parameters as SAPParameters,
       setTotalPages
     );
     setProducts(products);
     setErrors(errors);
-  }, [baseSite, query, page, sdk.parameters]);
+  }, [baseSite, debouncedQuery, page, sdk.parameters]);
+
+  useEffect(() => {
```

Comment:
I broke this out so we now have two `useEffect` hooks, which is part of resolving the issue with being able to change the basesite. 

What was happening is that our original `useEffect` hook fetched all of the basesites and set our state value for `baseSite` to whatever was at the first index of the basesites list. Then the hook would call `load` which fetches the products. Because of that, `load` was a dependency for the `useEffect` hook and anytime we called `load` we wound up triggering the `useEffect` hook. That caused the `baseSite` state value to get overwritten. 

We needed to separate that logic so that we only fetch/set the basesites on mount. The `load` function is a callback and it has the `baseSite` as a dependency so we don't need to explicitly call `load` again here after we initially fetch those baseSites.

That's also why I was able to remove the `load` calls elsewhere in the file. Overall I believe we were fetching the product list a lot more often than necessary.

Reply:
very cool, appreciate the write up